autoload/coverage/lcov.vim

```diff
@@ -0,0 +1,211 @@
+" Copyright 2021 Google Inc. All rights reserved.
+"
+" Licensed under the Apache License, Version 2.0 (the "License");
+" you may not use this file except in compliance with the License.
+" You may obtain a copy of the License at
+"
+"     http://www.apache.org/licenses/LICENSE-2.0
+"
+" Unless required by applicable law or agreed to in writing, software
+" distributed under the License is distributed on an "AS IS" BASIS,
+" WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+" See the License for the specific language governing permissions and
+" limitations under the License.
+
+"{{{ Init
+
+let s:plugin = maktaba#plugin#Get('coverage')
+
+"}}}
+
+"{{{ [lcov](https://github.com/linux-test-project/lcov) coverage provider.
+
+""
+" @private
+" Gets a list of files matching the plugin settings.
+function! s:GetCoverageDataPaths() abort
+  let l:paths = join(s:plugin.Flag('lcov_search_paths'), ',')
```

Comment:
I'm considering whether a maktaba flag is a good fit for this since it's global and can't handle any per-path or per-invocation distinctions… We've discussed one case with bazel coverage where a global list of paths isn't a great fit. What other common cases will there be for configuring this?

A flag _could_ be worth having if there are good common use cases for a global search path you can customize, but if so I'd like to make sure it works intuitively and maybe has more explicit validation that you're indeed passing it a list of things that do look like paths.

Then later we'd want something to support the more dynamic overrides. Possible mechanisms that could handle that:
  * A flag that accepts a function instead of just a static list
  * A `b:lcov_search_paths` variable to override search path per buffer
  * An options dict you can pass through `:CoverageShow` / `coverage#Show` for per-invocation overrides
  * Some other extension mechanism that other plugins could hook into

Reply:
To unblock the `bazel coverage` prototyping in the short term, let's just use a one-off solution that's clearly temporary. You could stash some data in `maktaba#plugin#Get('coverage').globals`, something like `._lcov_temp_search_path`. Let's handle that as a separate PR though so we can keep this one clean and focused on the basics of getting a usable lcov implementation.

Reply:
Makes sense.  I think in general some configuration will be needed; there doesn't seem to be a clear "sensible default" for the search path, as there doesn't seem to be a standard place to find gcov output.  Some projects go out of their way to place it in a specific directory with an arbitrary name (coverage, _coverage, gcovcoverage, ...), and some leave it next to the binaries that produced it (which could be spread across multiple directories, totally distinct from the source files).

I think a per-invocation solution would be the most robust for bazel (because the output path doesn't actually exist until the coverage command is run - i.e., it may not exist when vim or even the buffer is loaded, and is not determined by attributes of the buffer) I just want to make sure that the configuration isn't too cumbersome, as it seems like more often than not users will need to configure it for their project/workflow.

but, I'll put this aside for now and focus pick it up in #42.